TeamCode/src/main/java/org/firstinspires/ftc/teamcode/tests/JUnitTests.java

```diff
@@ -0,0 +1,115 @@
+package org.firstinspires.ftc.teamcode.tests;
+
+import com.acmerobotics.roadrunner.geometry.Pose2d;
+import com.acmerobotics.roadrunner.geometry.Vector2d;
+import com.acmerobotics.roadrunner.trajectory.Trajectory;
+import com.acmerobotics.roadrunner.trajectory.TrajectoryBuilder;
+import com.qualcomm.robotcore.hardware.DcMotor;
+import com.qualcomm.robotcore.hardware.DcMotorEx;
+
+import org.firstinspires.ftc.teamcode.components.Encoder;
+import org.firstinspires.ftc.teamcode.components.IntakeSystem;
+import org.firstinspires.ftc.teamcode.components.RoadRunnerDriveSystem;
+import org.firstinspires.ftc.teamcode.components.Tensorflow;
+import org.firstinspires.ftc.teamcode.helpers.GameState;
+import org.firstinspires.ftc.teamcode.helpers.Target;
+
+public class JUnitTests extends AutonomousSelfCheck {
+
+    private Pose2d oldPosEstimate;
+    private IntakeSystem intake = new IntakeSystem(hardwareMap.get(DcMotor.class, "IntakeSystem"));
+    private boolean first = true;
+    TrajectoryBuilder trajectoryBuilder = RoadRunnerDriveSystem.trajectoryBuilder(currentPosition);
+    private RoadRunnerDriveSystem roadRunnerDriveSystem;
+
+    @Override
+    public void loop() {
+        double x = time;
+        int i = 0;
+        switch (currentGameState) { // TODO: This monstrosity.
+            //TODO Do we need a trajectory as a field?
+            case TEST_ENCODERS:
+                boolean bool = true;
+                for (DcMotorEx motors : roadRunnerDriveSystem.getMotors()) {
+                    Encoder encoder = new Encoder(motors);
+                    if (encoder.getCorrectedVelocity() != 0) {
+                        bool = false;
+                    }
+                    if (encoder.getCorrectedVelocity() != 0) {
+                        bool = false;
+                    }
+                    Encoder.Direction direction = encoder.getDirection();
+                    encoder.setDirection(encoder.getDirection());
+                    if (direction != encoder.getDirection()) {
+                        bool = false;
+                    }
+                    // if (encoderTicksToInches == ) TODO: we need a way to check that our conversions are correct
+                    if (bool = false) {
```

Comment:
I detect that this code is problematic. According to the [Correctness (CORRECTNESS)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#correctness-correctness), [QBA: Method assigns boolean literal in boolean expression (QBA_QUESTIONABLE_BOOLEAN_ASSIGNMENT)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#qba-method-assigns-boolean-literal-in-boolean-expression-qba-questionable-boolean-assignment).
This method assigns a literal boolean value (true or false) to a boolean variable inside an if or while expression. Most probably this was supposed to be a boolean comparison using ==, not an assignment using =.